HenBot/UrlExtractor.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace HenBot
+{
+    public class UrlExtractor
+    {
+        public static List<string> ExtractUrlsFromPostsList(List<Post> postsList)
+        {
+            var urls = new List<string>(postsList.Capacity);
+            foreach (var post in postsList)
+            {
+                if (post.Sample == 0)
+                {
+                    urls.Add(post.File_Url);
+                }
+                else
+                {
+                    urls.Add(post.Sample_Url);
+                }
+            }
+
+            return urls;
+        }
+    }
```

Comment:
LINQ is a really good choice for doing this

```suggestion
public static List<string> ExtractUrlsFromPostsList(List<Post> postsList)
        {
            return postsList
                .Select(p => p.Sample == 0 ? p.File_Url : p.Sample_Url)
                .ToList();
        }
```